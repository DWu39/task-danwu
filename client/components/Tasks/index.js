import PropTypes from 'prop-types';
import React from 'react';

import TaskList from './TaskList';
import Toolbar from './Toolbar';

const Tasks = ({
  closeTask,
  deleteTask,
  edited,
  newTask,
  openTask,
  saveTasks,
  tasks
}) => (
  <section>
    <Toolbar
      edited={edited}
      newTask={newTask}
      saveTasks={() => { saveTasks(tasks); }}
    />
    <TaskList
      closeTask={closeTask}
      deleteTask={deleteTask}
      openTask={openTask}
      tasks={tasks}
    />
  </section>
);

Tasks.propTypes = {
  closeTask: PropTypes.func,
  deleteTask: PropTypes.func,
  edited: PropTypes.bool,
  newTask: PropTypes.func,
  openTask: PropTypes.func,
  saveTasks: PropTypes.func,
  tasks: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string.isRequired,
      text: PropTypes.string.isRequired
    })
  )
}

Tasks.defaultProps = {
  closeTask: null,
  deleteTask: null,
  edited: false,
  newTask: null,
  openTask: null,
  saveTasks: null,
  tasks: []
}

export default Tasks;
