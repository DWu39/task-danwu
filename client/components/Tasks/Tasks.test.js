import React from 'react';
import { shallow } from 'enzyme';

import Tasks from './index';

describe('Tasks', () => {
  it('should render correctly', () => {
    const output = shallow(<Tasks>Press</Tasks>);
    expect(output).toMatchSnapshot();
  });
});
