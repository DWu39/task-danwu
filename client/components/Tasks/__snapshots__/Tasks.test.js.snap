// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tasks should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Tasks
    closeTask={null}
    deleteTask={null}
    edited={false}
    newTask={null}
    openTask={null}
    saveTasks={null}
    tasks={Array []}
  >
    Press
  </Tasks>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Toolbar
          edited={false}
          newTask={null}
          saveTasks={null}
        />,
        <TaskList
          closeTask={null}
          deleteTask={null}
          openTask={null}
          tasks={Array []}
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "edited": false,
          "newTask": null,
          "saveTasks": null,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "closeTask": null,
          "deleteTask": null,
          "openTask": null,
          "tasks": Array [],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Toolbar
            edited={false}
            newTask={null}
            saveTasks={null}
          />,
          <TaskList
            closeTask={null}
            deleteTask={null}
            openTask={null}
            tasks={Array []}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "edited": false,
            "newTask": null,
            "saveTasks": null,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "closeTask": null,
            "deleteTask": null,
            "openTask": null,
            "tasks": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
